Packs/Thycotic/Integrations/Thycotic/Thycotic.py

```diff
@@ -0,0 +1,531 @@
+    user = client.userCreate(**kwargs)
+
+    markdown = tableToMarkdown('Created new user', user)
+
+    return CommandResults(
+        readable_output=markdown,
+        outputs_prefix="Thycotic.User.Create",
+        outputs_key_field="user",
+        raw_response=user,
+        outputs=user
+    )
+
+
+def user_search_command(client, **kwargs):
+    user = client.userSearch(**kwargs)
+    markdown = tableToMarkdown('Search user', user)
+
+    return CommandResults(
+        readable_output=markdown,
+        outputs_prefix="Thycotic.User.Search",
+        outputs_key_field="user",
+        raw_response=user,
+        outputs=user
+    )
+
+
+def user_update_command(client, id: str = '', **kwargs):
+    user = client.userUpdate(id, **kwargs)
+    markdown = tableToMarkdown('Updated user', user)
+
+    return CommandResults(
+        readable_output=markdown,
+        outputs_prefix="Thycotic.User.Update",
+        outputs_key_field="user",
+        raw_response=user,
+        outputs=user
+    )
+
+
+def user_delete_command(client, id: str = ''):
+    user = client.userDelete(id)
+    markdown = tableToMarkdown('Deleted user', user)
+
+    return CommandResults(
+        readable_output=markdown,
+        outputs_prefix="Thycotic.User.Delete",
+        outputs_key_field="user",
+        raw_response=user,
+        outputs=user
+    )
+
+
+def secret_rpc_changepassword_command(client, secret_id: str = '', newpassword: str = ''):
+    secret = client.secretChangePassword(secret_id, newpassword)
+    markdown = tableToMarkdown('Change password for remote machine', secret)
+
+    return CommandResults(
+        readable_output=markdown,
+        outputs_prefix="Thycotic.Secret.ChangePassword",
+        outputs_key_field="secret",
+        raw_response=secret,
+        outputs=secret
+    )
+
+
+def main():
+    username = demisto.params().get('credentials').get('identifier')
+    password = demisto.params().get('credentials').get('password')
+
+    # get the service API url
+    url = demisto.params().get('url')
+    proxy = demisto.params().get('proxy', False)
+    verify = not demisto.params().get('insecure', False)
+
+    LOG(f'Command being called is {demisto.command()}')
+
+    thycotic_commands = {
+        'thycotic-secret-password-get': secret_password_get_command,
+        'thycotic-secret-username-get': secret_username_get_command,
+        'thycotic-secret-get': secret_get_command,
+        'thycotic-secret-search-name': secret_search_name_command,
+        'thycotic-secret-search': secret_search_command,
+        'thycotic-secret-password-update': secret_password_update_command,
+        'thycotic-secret-checkout': secret_checkout_command,
+        'thycotic-secret-checkin': secret_checkin_command,
+        'thycotic-secret-create': secret_create_command,
+        'thycotic-secret-delete': secret_delete_command,
+        'thycotic-secret-rpc-changepassword': secret_rpc_changepassword_command,
```

Comment:
What does this command do?

Reply:
@guykeller 
**checkout/checkin**


Secret Server’s Check Out feature forces accountability by granting exclusive access to a single user with a One Time Password (OTP). OTP, also known as one-time pin or dynamic password, is a password that is valid for only one login session or transaction. OTPs avoid a number of risks associated with traditional, static passwords, while also provide the critical privileged access that certain users require to perform a specific task or function.

After ‘Check In,’ Secret Server automatically performs a random password change on the remote machine. No other user can access a secret while it’s checked out. This requirement establishes direct accountability for a machine that’s accessed during a specific time period.

Check Out ensures that Secret Server is used for all password changes and generates a complete audit trail of password use.

Advanced scripting capabilities allow Check Out to integrate with PowerShell hooks that can run before and after a secret is checked out. These hooks can be used for custom checks or actions to ensure environments are ready, or validate that specific conditions are met, before a user can access a secret.

**thycotic-secret-rpc-changepassword - Remote Passwoed Changing**
allows you to change the password on a remote machine with different policies